internal/app/udevd/main.go

```diff
@@ -10,50 +10,45 @@ import (
 	"fmt"
 	"log"
 	"os"
-	"path"
+	"path/filepath"
 
+	"github.com/mdlayher/kobject"
 	"github.com/pkg/errors"
 	"github.com/talos-systems/talos/internal/app/udevd/internal/drivers/scsi"
-	"github.com/talos-systems/talos/internal/app/udevd/internal/uevent"
 )
 
 func init() {
 	log.SetFlags(log.Lshortfile | log.Ldate | log.Lmicroseconds | log.Ltime)
 }
 
 func watch() error {
-	k, err := uevent.Dial()
+	client, err := kobject.New()
 	if err != nil {
 		return err
 	}
-	// nolint: errcheck
-	defer k.Close()
-
-	for u := range k.Watch() {
-		if u.Error != nil {
-			log.Printf("uevent error: %+v\n", err)
-			continue
+	for {
+		event, err := client.Receive()
+		if err != nil {
+			log.Printf("failed to receive event: %v", err)
```

Comment:
You'll need a continue after this log.

Reply:
Crazy how sleep affects you when you don’t have it 😄. My newborn has been keeping me up all day every day. Thanks for catching.

Reply:
All good!